roofit/roofitcore/inc/RooArgSet.h

```diff
@@ -36,50 +36,41 @@ class RooArgSet : public RooAbsCollection {
-	    const RooAbsArg& var3, 
-	    const char *name="");
-  RooArgSet(const RooAbsArg& var1, const RooAbsArg& var2,
-	    const RooAbsArg& var3, const RooAbsArg& var4, 
-	    const char *name="");
-  RooArgSet(const RooAbsArg& var1, const RooAbsArg& var2,
-	    const RooAbsArg& var3, const RooAbsArg& var4, 
-	    const RooAbsArg& var5, 
-	    const char *name="");
-  RooArgSet(const RooAbsArg& var1, const RooAbsArg& var2,
-	    const RooAbsArg& var3, const RooAbsArg& var4, 
-	    const RooAbsArg& var5, const RooAbsArg& var6, 
-	    const char *name="");
-  RooArgSet(const RooAbsArg& var1, const RooAbsArg& var2,
-            const RooAbsArg& var3, const RooAbsArg& var4, 
-	    const RooAbsArg& var5, const RooAbsArg& var6, 
-	    const RooAbsArg& var7, 
-	    const char *name="");
-  RooArgSet(const RooAbsArg& var1, const RooAbsArg& var2,
-            const RooAbsArg& var3, const RooAbsArg& var4, 
-	    const RooAbsArg& var5, const RooAbsArg& var6, 
-	    const RooAbsArg& var7, const RooAbsArg& var8, 
-	    const char *name="");
-  RooArgSet(const RooAbsArg& var1, const RooAbsArg& var2,
-            const RooAbsArg& var3, const RooAbsArg& var4, 
-	    const RooAbsArg& var5, const RooAbsArg& var6, 
-	    const RooAbsArg& var7, const RooAbsArg& var8, 
-	    const RooAbsArg& var9, const char *name="");
+
+private:
+  // Helper to check the types of the variadic template below:
+  template<bool...> struct bool_pack;
+  template<bool... bs>
+  using all_true = std::is_same<bool_pack<bs..., true>, bool_pack<true, bs...>>;
+public:
+  /// Construct a (non-owning) RooArgSet from one or more
+  /// RooFit objects or collections. The set will not own its contents.
+  /// \tparam T A RooFit class that derives from RooAbsArg or a RooFit collection.
+  /// \tparam Ts Parameter pack of objects that derive from RooAbsArg or RooFit collections; or a name.
+  /// \param arg A RooFit object that derives from RooAbsArg, or a collection of RooFit objects.
+  /// \param moreArgsOrName Arbitrary number of
+  /// - Further RooFit objects that derive from RooAbsArg
+  /// - RooFit collections of such objects
+  /// - A name for the set. Given multiple names, the last-given name prevails.
+  template<typename T, typename... Ts,
+      typename = typename std::enable_if<std::is_base_of<RooAbsArg, T>::value || std::is_base_of<RooAbsCollection, T>::value, void>::type,
+      typename = typename std::enable_if<all_true<
+      std::is_base_of<RooAbsArg, Ts>::value
+      || std::is_base_of<RooAbsCollection, Ts>::value
+      || std::is_convertible<Ts, const char*>::value ...>::value, void>::type >
+  RooArgSet(const T& arg, const Ts&... moreArgsOrName)
+  /* NB: Making this a delegating constructor lead to linker errors with MSVC,
+   * so no calling of the empty constructor here.*/ {
+    processArg(arg);
+    // Expand parameter pack in C++ 11 way:
+    int dummy[] = { 0, ( (void) processArg(moreArgsOrName), 0) ... };
```

Comment:
I *think* I read that that `(void)` was necessary to suppress an unused argument warning if something with the template failed. Can you find the original stack post overflow again?

Reply:
Not sure what stackoverflow post you are referring to, but I never had to add it whenever I used the trick in RDF. I think it's extra noise until proven otherwise (but not blocking a merge if you really want it :smile: )

Reply:
For future reference: it's [this](https://stackoverflow.com/questions/59590426/variadic-template-argument-forwarding-uses-comma-operator), and it's only needed if somebody redefines the comma operator :flushed: